deebotozmo/cli.py

```diff
@@ -0,0 +1,458 @@
+    finally:
+        await vacbot.after()
+
+
+@cli.command(name="getrooms", help="Get saved rooms")
+@coro
+async def get_rooms() -> None:
+    """Click subcommand that returns saved room."""
+    vacbot = CliUtil()
+    try:
+        await vacbot.before()
+
+        lock = asyncio.Event()
+
+        async def on_rooms(event: RoomsEvent) -> None:
+            for room in event.rooms:
+                print(f"{room.id} {room.subtype}")
+            lock.set()
+
+        lock.set()
+        listener = vacbot.bot.map.events.rooms.subscribe(on_rooms)
+        await lock.wait()
+        listener.unsubscribe()
+    finally:
+        await vacbot.after()
+
+
+@cli.command(
+    name="exportlivemap",
+    help='Get robot map and save it [filepath ex: "/folder/livemap.png"',
+)
+@click.argument("filepath", type=click.STRING, required=True)
+@click.option(
+    "--force-extension/--no-force-extension",
+    default=False,
+    help="Do not check if the file extension is valid.",
+)
+@coro
+async def export_live_map(filepath: str, force_extension: bool) -> None:
+    """Click subcommand that returns the live map."""
+    if not force_extension and mimetypes.guess_type(filepath)[0] != "image/png":
+        logging.error("exportlivemap generates a png image.")
+        logging.error(
+            "either change your file extension to 'png' or pass '--force_extension'."
+        )
+        return
+
+    vacbot = CliUtil()
+    try:
+        await vacbot.before()
+
+        lock = asyncio.Event()
+
+        async def on_map(_: MapEvent) -> None:
+            with open(filepath, "wb") as file:
+                file.write(base64.decodebytes(vacbot.bot.map.get_base64_map()))
+            lock.set()
+
+        lock.clear()
+        listener = vacbot.bot.events.map.subscribe(on_map)
+        await lock.wait()
+        listener.unsubscribe()
+    finally:
+        await vacbot.after()
+
+
+class CliUtil:
+    """CliUtil communicates with the bot to provide self.bot."""
+
+    def __init__(self) -> None:
+        """Init CliUtil."""
+        self._config = read_config()
+
+        self._session = aiohttp.ClientSession()
+
+        self._api = EcovacsAPI(
+            self._session,
+            self._config["device_id"],
+            self._config["email"],
+            self._config["password_hash"],
+            continent=self._config["continent"],
+            country=self._config["country"],
+            verify_ssl=bool(self._config["verify_ssl"]),
+        )
+
+        if not config_file_exists():
+            click.echo(
+                f"Not logged in. Do '{os.path.basename(sys.argv[0])} createconfig' first."
+            )
+            sys.exit(1)
+
+        return
+
+    async def before(self) -> None:
+        # pylint: disable=attribute-defined-outside-init
+        """Communicate with Deebot."""
+        await self._api.login()
+
+        self.devices_ = await self._api.get_devices()
+
+        self.auth = await self._api.get_request_auth()
```

Comment:
```suggestion
        auth = await self._api.get_request_auth()
```
Can be local variable